### PR TITLE
Bugfix/superlu dist constructor

### LINKED ISSUE #169: SuperLU_DIST matrix interface constructor definition does not match declaration
https://github.com/LLNL/sundials/blob/e2f29c34f324829302037a1492db480be8828084/src/sunmatrix/slunrloc/sunmatrix_slunrloc.c#L79

### PR DESCRIPTION
**Pull request by balos1**
Fixes #169

### CODE REVIEW

**Review by drreynolds — DISMISSED** [review]

File: test/env/docker.sh [hunk 1]
```diff
@@ -276,20 +276,20 @@ fi
...
 if [ "$SUNDIALS_PRECISION" == "double" ]; then
     export SUNDIALS_SUPERLU_DIST=ON
-    export SUPERLU_DIST_ROOT=/opt/view
+    export SUPERLU_DIST_ROOT=/opt/view/superlu-dist
     export SUPERLU_DIST_INCLUDE_DIR="${SUPERLU_DIST_ROOT}/include"
     export SUPERLU_DIST_LIBRARY_DIR="${SUPERLU_DIST_ROOT}/lib"
 
     # built with 32-bit blas
-    export BLAS_ROOT=/opt/view
+    export BLAS_ROOT=/opt/view/superlu-dist
```
Comment (gardner48) [inline test/env/docker.sh]: I think this and and lines below are the causing the CI failures
```suggestion
    export BLAS_ROOT=/opt/view
```

File: test/env/docker.sh [hunk 2]
```diff
@@ -276,20 +276,20 @@ fi
...
     export BLAS_LIBRARIES=${BLAS_ROOT}/lib/libopenblas.so
 
     # PARMETIS
-    export PARMETIS_ROOT=/opt/view
+    export PARMETIS_ROOT=/opt/view/superlu-dist
```
Comment (gardner48) [inline test/env/docker.sh]:
```suggestion
    export PARMETIS_ROOT=/opt/view
```

File: test/env/docker.sh [hunk 3]
```diff
@@ -276,20 +276,20 @@ fi
...
     export PARMETIS_LIBRARIES="${PARMETIS_ROOT}/lib/libparmetis.so"
 
     # METIS
-    export METIS_ROOT=/opt/view
+    export METIS_ROOT=/opt/view/superlu-dist
```
Comment (gardner48) [inline test/env/docker.sh]:
```suggestion
    export METIS_ROOT=/opt/view
```

**Review by gardner48 — APPROVED** [review]

File: src/sunmatrix/slunrloc/sunmatrix_slunrloc.c [hunk 4]
```diff
@@ -76,7 +76,7 @@ static booleantype SMCompatible_SLUNRloc(SUNMatrix A, SUNMatrix B);
  * ----------------------------------------------------------------------------
  */
 
-SUNMatrix SUNMatrix_SLUNRloc(SuperMatrix *A_super, gridinfo_t *grid)
+SUNMatrix SUNMatrix_SLUNRloc(SuperMatrix *A_super, gridinfo_t *grid, SUNContext sunctx)
 {
   SUNMatrix A;
   SUNMatrix_Ops ops;
```
Comment (gardner48) [inline src/sunmatrix/slunrloc/sunmatrix_slunrloc.c:82]:
```suggestion
```

File: src/sunmatrix/slunrloc/sunmatrix_slunrloc.c [hunk 5]
```diff
@@ -91,16 +91,11 @@ SUNMatrix SUNMatrix_SLUNRloc(SuperMatrix *A_super, gridinfo_t *grid)
...
+  A = SUNMatNewEmpty(sunctx);
   if (A == NULL) return(NULL);
 
-  /* Create the matrix operations structure */
-  ops = NULL;
-  ops = (SUNMatrix_Ops) malloc(sizeof(struct _generic_SUNMatrix_Ops));
-  if (ops == NULL) { free(A); return(NULL); }
-
   /* Attach operations */
   ops->getid       = SUNMatGetID_SLUNRloc;
```
Comment (gardner48) [inline src/sunmatrix/slunrloc/sunmatrix_slunrloc.c:100]: Here and below `A->ops`

**Comment by balos1** [comment]
Closing this PR and I will open a new one that also has support for newer SuperLU_DIST and GPUs.

**Pull request closed without merging**